sdk/identity/azure-identity/inc/azure/identity/chained_token_credential.hpp

```diff
@@ -0,0 +1,62 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// SPDX-License-Identifier: MIT
+
+/**
+ * @file
+ * @brief Chained Token Credential.
+ */
+
+#pragma once
+
+#include <azure/core/credentials/credentials.hpp>
+
+#include <memory>
+#include <vector>
+
+namespace Azure { namespace Identity {
+  /**
+   * @brief Chained Token Credential provides a token credential implementation which chains
+   * multiple `%Core::Credentials::TokenCredential` implementations to be tried in order until one
+   * of the `%GetToken()` methods returns an access token.
+   *
+   */
+  class ChainedTokenCredential final : public Core::Credentials::TokenCredential {
+  public:
+    /**
+     * @brief A container type to store the ordered chain of credentials.
+     *
+     */
+    using Sources = std::vector<std::shared_ptr<Core::Credentials::TokenCredential>>;
+
+  private:
+    Sources m_sources;
+
+  public:
```

Comment:
nit: can we avoid multiple private/public categories, and only have just one?

Reply:
It is possible, especially that we don't have inline functions defined in the class body. I'll do the change.